pkg/ctl/install/pki.go

```diff
@@ -0,0 +1,167 @@
+package install
+
+import (
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"github.com/cloudflare/cfssl/cli/genkey"
+	"github.com/cloudflare/cfssl/config"
+	"github.com/cloudflare/cfssl/csr"
+	"github.com/cloudflare/cfssl/helpers"
+	"github.com/cloudflare/cfssl/initca"
+	"github.com/cloudflare/cfssl/log"
+	"github.com/cloudflare/cfssl/signer"
+	"github.com/cloudflare/cfssl/signer/local"
+)
+
+type publicKeyInfrastructurePaths struct {
+	caCertificate     string
+	caKey             string
+	serverCertificate string
+	serverKey         string
+	clientCertificate string
+	clientKey         string
+}
+
+type publicKeyInfrastructure struct {
```

Comment:
These structs seem a little too verbose. Why not have just a map or a single struct?

Reply:
Because of their use. They can be used separately. Please suggest a specific change if you think it can be done differently.

Reply:
Resolving for now. Please unresolve it if you find a nicer design.